src/diamond/handler/influxdbHandler.py

```diff
@@ -159,39 +176,63 @@ def _send(self):
         """
         # Check to see if we have a valid socket. If not, try to connect.
         try:
-            if self.influx is None:
-                self.log.debug("InfluxdbHandler: Socket is not connected. "
-                               "Reconnecting.")
-                self._connect()
-            if self.influx is None:
-                self.log.debug("InfluxdbHandler: Reconnect failed.")
-            else:
-                # build metrics data
-                metrics = []
-                for path in self.batch:
-                    metrics.append({
-                        "points": self.batch[path],
-                        "name": path,
-                        "columns": ["time", "value"]})
-                # Send data to influxdb
-                self.log.debug("InfluxdbHandler: writing %d series of data",
-                               len(metrics))
-                self.influx.write_points(metrics,
-                                         time_precision=self.time_precision)
-
-                # empty batch buffer
-                self.batch = {}
-                self.batch_count = 0
-                self.time_multiplier = 1
+                if self.influx is None:
+                    self.log.debug("InfluxdbHandler: Socket is not connected. "
+                                   "Reconnecting.")
+                    self._connect()
+                if self.influx is None:
+                    self.log.debug("InfluxdbHandler: Reconnect failed.")
+                else:
+                    # build metrics data
+                    metrics = []
+                    if self.influxdb_version == "0.8":
+                        for path in self.batch:
+                            metrics.append({
+                                "points": self.batch[path],
+                                "name": path,
+                                "columns": ["time", "value"]})
+                    elif self.influxdb_version == "0.9":
+                        for path in self.batch:
+                            # split path to use influx syntax for tags and tables.
+                            tokenized = path.split('.')
```

Comment:
Instead of accessing this with indexed keys, ie `tokenized[2]`, I think it would be good to assign them to variables that more clearly explain what you believe the field holds.